response_operations_ui/views/respondents.py

```diff
@@ -50,4 +50,91 @@ def respondent_details(respondent_id):
+def edit_contact_details(respondent_id):
+    form = request.form
+    contact_details_changed = party_controller.update_contact_details(respondent_id, form)
+
+    ui_message = 'No updates were necessary'
+    if 'emailAddress' in contact_details_changed:
+        ui_message = f'Contact details changed and verification email sent to {form.get("email")}'
+    elif len(contact_details_changed) > 0:
+        ui_message = 'Contact details changed'
+
+    return redirect(url_for('respondent_bp.respondent_details', respondent_id=respondent_id, info=ui_message))
+
+
+@respondent_bp.route('/resend_verification/<respondent_id>', methods=['GET'])
+@login_required
+def view_resend_verification(respondent_id):
+    logger.debug("Re-send verification email requested", respondent_id=respondent_id)
+    respondent = party_controller.get_respondent_by_party_id(respondent_id)
+    email = respondent['pendingEmailAddress'] if 'pendingEmailAddress' in respondent \
+        else respondent['emailAddress']
+
+    return render_template('re-send-verification-email.html', respondent_id=respondent_id, email=email,
+                           tab='respondents')
+
+
+@respondent_bp.route('/resend_verification/<party_id>', methods=['POST'])
+@login_required
+def resend_verification(party_id):
+    reporting_units_controllers.resend_verification_email(party_id)
+    logger.info("Re-sent verification email.", party_id=party_id)
+    return redirect(url_for('respondent_bp.respondent_details', respondent_id=party_id,
+                            info='Verification email re-sent'))
+
+
+@respondent_bp.route('<respondent_id>/change-enrolment-status', methods=['POST'])
+@login_required
+def change_enrolment_status(respondent_id):
+
+    reporting_units_controllers.change_enrolment_status(business_id=request.args['business_id'],
+                                                        respondent_id=respondent_id,
+                                                        survey_id=request.args['survey_id'],
+                                                        change_flag=request.args['change_flag'])
+    return redirect(url_for('respondent_bp.respondent_details', respondent_id=respondent_id,  enrolment_changed='True'))
+
+
+@respondent_bp.route('/<ru_ref>/change-enrolment-status', methods=['GET'])
+@login_required
+def confirm_change_enrolment_status(ru_ref):
+    return render_template('confirm-enrolment-change.html', business_id=request.args['business_id'], ru_ref=ru_ref,
+                           ru_name=request.args.get('ru_name'),
+                           trading_as=request.args['trading_as'], survey_id=request.args['survey_id'],
+                           survey_name=request.args['survey_name'], respondent_id=request.args['respondent_id'],
+                           first_name=request.args['respondent_first_name'],
+                           last_name=request.args['respondent_last_name'],
+                           change_flag=request.args['change_flag'],
```

Comment:
These functions you've added need to be tested - so I'm assuming that you're covering their testing with acceptance tests repo, because they aren't very pure, and thus not hugely testable directly.

If there isn't testing for this functionality at all, it needs adding.